TLM/TLM/Custom/PathFinding/CustomPathFind.cs

```diff
@@ -4088,15 +4098,41 @@ private struct BufferItem {
                 return true;
             }
 
-            ExtVehicleType allowedTypes = vehicleRestrictionsManager.GetAllowedVehicleTypes(
+            ExtVehicleType allowedPathVehicleTypes = allowMixedCargoPath_
+                                                        ? mixedPathVehicleTypes_
+                                                        : queueItem_.vehicleType;
```

Comment:
it seems to me if `allowMixedCargoPath_ `is false then `mixedPathVehicleTypes_  == queueItem_.vehicleType` because of this code from earlier on:
```cs
if (allowMixedCargoPath_) {
  mixedPathVehicleTypes_ = VehicleTypeToMixedCargoExtVehicle(vehicleTypes_, queueItem_.vehicleType);
}
```
therefore this line can be simplified to:
```suggestion
            ExtVehicleType allowedPathVehicleTypes = mixedPathVehicleTypes_;
```
maybe also rename `mixedPathVehicleTypes_ `to `allowedPathVehicleTypes_` or add a comment.

Reply:
The main reason for that is to help understanding that is specific behavior, applied only for specific vehicles in specific condition. If generalize it, it will become even more confusing if you back to the code after some time.